specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2018-06-01-preview/scriptActions.json

```diff
@@ -0,0 +1,420 @@
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ClusterNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ScriptNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK response definition."
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/executeScriptActions": {
+            "post": {
+                "tags": [
+                    "Clusters"
+                ],
+                "operationId": "Clusters_ExecuteScriptActions",
+                "description": "Executes script actions on the specified HDInsight cluster.",
+                "x-ms-examples": {
+                    "Execute script action on HDInsight cluster": {
+                        "$ref": "./examples/PostExecuteScriptAction.json"
+                    }
+                },
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ClusterNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "name": "parameters",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "./cluster.json#/definitions/ExecuteScriptActionParameters"
+                        },
+                        "description": "The parameters for executing script actions."
+                    }
+                ],
+                "responses": {
+                    "202": {
+                        "description": "OK response definition."
+                    },
+                    "200": {
+                        "description": "Accepted response definition."
+                    }
+                },
+                "x-ms-long-running-operation": true
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/scriptActions": {
+            "get": {
+                "tags": [
+                    "ScriptActions"
+                ],
+                "operationId": "ScriptActions_ListPersistedScripts",

```

Comment:
>ScriptActions_ListPersistedScripts [](start = 32, length = 34)

I think it would be better to name this operation as "ScriptActions_ListPersistedScriptsByCluster"

Reply:
BTW, do we need to specify that this is PresistedScripts in the operation? Wouldn't it be enough to just call this operation as "ScriptActions_ListByCluster" ?

---
In reply to: [197284866](https://github.com/Azure/azure-rest-api-specs/pull/3256#discussion_r197284866) [](ancestors = 197284866)

Reply:
I'll rename it the operation, but I'll keep it in the description. Persisted is an extraneous keyword.